src/index.ts

```diff
@@ -14,6 +15,7 @@ appInsights.setup(clientKey)
   .setUseDiskRetryCaching(true)
   .setSendLiveMetrics(process.env.AI_ENABLE_LIVEMETRICS === 'true') // Default to disabled: https://github.com/microsoft/ApplicationInsights-node.js/issues/615
   .setDistributedTracingMode(appInsights.DistributedTracingModes.AI_AND_W3C)
+appInsights.defaultClient.config.httpsAgent = new Agent({ keepAlive: true, maxSockets: 400 })
```

Comment:
We would need an env var override for maxSockets and a much higher default to be safe.

Reply:
added